connectors/sources/generic_database.py

```diff
@@ -362,27 +376,43 @@ async def fetch_rows(self, schema=None):
         Yields:
             Dict: Row document to index
         """
-        # Query to get all table names from a database
-        list_of_tables = await anext(
-            self.execute_query(
-                query_name="ALL_TABLE",
-                user=self.user.upper(),
-                database=self.database,
+        tables_to_fetch = await self.get_tables_to_fetch(schema)
+
+        for table in tables_to_fetch:
+            logger.debug(f"Found table: {table} in database: {self.database}.")
+            async for row in self.fetch_documents(
+                table=table,
                 schema=schema,
-            )
-        )
-        if len(list_of_tables) > 0:
-            for [table_name] in list_of_tables:
-                logger.debug(f"Found table: {table_name} in database: {self.database}.")
-                async for row in self.fetch_documents(
-                    table=table_name,
-                    schema=schema,
-                ):
-                    yield row
+            ):
+                yield row
         else:
             if schema:
                 logger.warning(
                     f"Fetched 0 tables for schema: {schema} and database: {self.database}"
                 )
             else:
                 logger.warning(f"Fetched 0 tables for the database: {self.database}")
+
+    async def get_tables_to_fetch(self, schema):
+        tables = configured_tables(self.tables)
+        fetch_all_tables = tables == ALL_TABLES or (
+            isinstance(tables, list) and tables == [ALL_TABLES]
+        )
+
+        tables_to_fetch = (
+            map(
+                lambda table: table[0],
+                await anext(
+                    self.execute_query(
+                        query_name="ALL_TABLE",
+                        user=self.user.upper(),
+                        database=self.database,
+                        schema=schema,
+                    )
+                ),
+            )
+            if fetch_all_tables
+            else tables
+        )
```

Comment:
I think it still makes sense to do some sort of validation of tables that user entered before trying to fetch data - what if they entered invalid table names, or ones that are not accessible by the user?

Reply:
I'll create a tech-debt issue for that and check, if we want to address/prioritise this on our side or simply let the datasource errors bubble up.

Reply:
I'm not quite sure about it because it changes logic a lot and should be delivered in same release this change is delivered at least.

Before this problem would not exist - if user specify correct database, tables cannot be incorrect, thus it's not possible to have database inconsistently synced (some tables yes, some tables not).

So to sum up, I think it's okay to handle it in separate PR, but I think it should happen in this release as well.

Reply:
Valid points, then we should also do the same validation for the database. Does it exist? is it accessible?

Created the issue for 8.8: https://github.com/elastic/enterprise-search-team/issues/4030

Reply:
For database it's a bit more lax cause if DB does not exist/can't be accessed, you'll get no data. For tables you'll get partial data which is worse than no data at all IMO